src/server/pps/server/monitor.go

```diff
@@ -249,49 +247,6 @@ func (pc *pipelineController) monitorPipeline(ctx context.Context, pipelineInfo
 			}, backoff.NewInfiniteBackOff(),
 				backoff.NotifyCtx(ctx, "monitorPipeline for "+pipeline))
 		})
-		if pipelineInfo.Details.ParallelismSpec != nil && pipelineInfo.Details.ParallelismSpec.Constant > 1 && pipelineInfo.Details.Autoscaling {
```

Comment:
Could you remind me what the main issue was that prompted moving this logic into scaleUpPipeline? Was it contention over the k8s API server with queries for the RC?

Reply:
At a high level, it's mostly to simplify the code, so that the only places where a pipeline RC is getting updated is by the pipeline's pipeline_controller, running in a synchronous loop. This sort of came out of the ["k8s stuckness" issue](https://github.com/pachyderm/pachyderm/pull/6038) from last year.

To explain "simplify the code" in a little more detail: when debugging that issue, one theory we had was that there was interference between `pollPipelines` and the autoscaling goro: if a pipeline was RUNNING and autoscaling was enabled, and then `step()` would set the number of pipeline replicas to 1, and if `pollPipelines` bumps a pipeline that's in the middle of running, that will drop its RC replicas down to 1 until the autoscaling goro runs and raises it again. That actually doesn't happen because `scaleUpPipeline()` only adjusts an RC's scale if it's currently at 0, but with the code written this way, the theory doesn't make sense because the RC isn't subject to any contention.

FWIW, I actually think the whole `monitorPipeline` goro could be inlined in the pipeline RC, making all pipeline updates synchronous. I actually think we could go a step farther and eliminate `STANDBY` completely, and have the handler for `RUNNING` (as well as `InspectPipeline`) just look at the pipeline's output repo for ready commits. Basically `RUNNING`+no ready commits=`STANDBY`, and then we wouldn't need a goro to explicitly change the pipeline's state. We'd just need a way to bump the pipelineController as soon as a new output commit is ready, but the poller would eventually fix things anyway in this model. All of this seemed out of scope for the change though.

Reply:
(also, if you think this change is a mistake, and we should limit this PR to eliminating no-op scaling updates, I'd be happy to do that)

Reply:
Ok that makes a lot of sense! I totally agree with the approach just wasn't seeing the literal contension issue I remember chatting about at a high level